syntax = "proto3";

package syft.core.node.common.service;

import "proto/syft_assets/common_object.proto";
import "proto/syft_assets/io/address.proto";


message SendAssociationRequestMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  string content = 3;
  syft.core.io.Address reply_to = 4;
}

message SendAssociationRequestResponse {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
}


message ReceiveAssociationRequestMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address reply_to = 4;
}

message ReceiveAssociationRequestResponse {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
}


message GetAssociationRequestMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address reply_to = 4;
}

message GetAssociationRequestResponse {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
}


message DeleteAssociationRequestMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address reply_to = 4;
}

message DeleteAssociationRequestResponse {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
}